src/client/CodeEditor/widgets.ts

```diff
@@ -312,3 +313,73 @@ class ColorWidget extends WidgetType {
     return false;
   }
 }
+
+export const highlightWidgets = ViewPlugin.fromClass(
+  class {
+    showHighlight: boolean;
+    view: EditorView;
+    constructor(view: EditorView) {
+      this.showHighlight = false;
+      this.view = view;
+    }
+  },
+  {
+    decorations: (v) => {
+      if (!v.showHighlight) {
+        return Decoration.none;
+      }
+
+      const interactOpportunities = [];
+      const lines = v.view.state.doc.iter();
+      let line = lines.next();
+
+      //Offset is the number of characters before the regex so the highlighting can be placed properly.
+      let offset = 0;
+      while (!line.done) {
+        if (line.value === '\n') {
+          offset++;
+          line = lines.next();
+          continue;
+        }
+        const interactiveOccurances = line.value.matchAll(
+          //The below line contains all of the Regexes for all of the interactive widgets.
+          /\"\#([0-9]|[A-F]|[a-f]){6}\"|rotate\(-?\d*\.?\d*\)|https?:\/\/[^ "]+|vec2\(-?\b\d+\.?\d*\b\s*(,\s*-?\b\d+\.?\d*\b)?\)|true|false|(?<!\#)-?\b\d+\.?\d*\b|rgb\(.*\)/dg,
```

Comment:
Very nice! One drawback I see here is that if one of these regexes needs to change for some reason, it needs to be changed here and also in the individual widgets. I wonder if there is a way to pull in the regex definitions from the other modules and combine them here programmatically...